src/Sys/InspectSeoHandler.php

```diff
@@ -0,0 +1,172 @@
+<?php
+declare(strict_types=1);
+
+namespace Triniti\Sys;
+
+use Gdbots\Ncr\Exception\NodeNotFound;
+use Gdbots\Ncr\Ncr;
+use Gdbots\Pbj\Message;
+use Gdbots\Pbjx\CommandHandler;
+use Gdbots\Pbjx\Pbjx;
+use Gdbots\UriTemplate\UriTemplateService;
+use Google\Service\SearchConsole\InspectUrlIndexResponse;
+use Psr\Log\LoggerInterface;
+use Triniti\Schemas\Sys\Event\SeoInspectedV1;
+
+class InspectSeoHandler implements CommandHandler
+{
+    public static function handlesCuries(): array
+    {
+        return [
+            'triniti:sys:command:inspect-seo',
+        ];
+    }
+
+    public function __construct(
+        private readonly Ncr        $ncr,
+        private readonly Flags      $flags,
+        private readonly LoggerInterface $logger,
+        private readonly array $config,
+    )
+    {
+    }
+
+    public function handleCommand(Message $command, Pbjx $pbjx): void
+    {
+        if ($this->flags->getBoolean('inspect_seo_handler_disabled') || !$command->has('search_engines')) {
+            return;
+        }
+
+        $searchEngines = $command->get('search_engines');
+        $nodeRef = $command->get('node_ref');
+
+        try {
+            $context = ['causator' => $command];
+            $node = $this->ncr->getNode($nodeRef, false, $context);
+        } catch (NodeNotFound $e) {
+            $this->logger->error('Unable to get node for search engines processing.', [
+                'exception' => $e,
+                'node_ref' => $nodeRef
+            ]);
+            return;
+        }
+
+        $retryCommand = clone $command;
+        $retryCommand->clear('search_engines');
+        $pbjx->copyContext($command, $retryCommand);
+
+        foreach ($searchEngines as $searchEngine) {
+            $methodName = 'checkIndexStatusFor' . ucfirst($searchEngine);
+
+            if (!method_exists($this, $methodName)) {
+                continue;
+            }
+
+            $retryCommand = $this->$methodName($retryCommand, $pbjx, $node);
+        }
+
+        if (empty($retryCommand->get('search_engines'))) {
+            return;
+        }
+
+        $retries = $command->get('ctx_retries');
+        $maxRetries = $this->flags->getInt('inspect_seo_max_retries', 5);
+
+        if ($retries < $maxRetries) {
```

Comment:
@digitalartcrew I see a check for max retries here and in the specific google method, I feel we should only need to do this once

Reply:
That is ideal but currently inside the inspection method we need a condition for adding a search engine to be retried then it is checked here: https://github.com/triniti/core-php/pull/34/files#diff-6ab784c5caf2e67b8d9e4a618627f1ab2974c5e93b92d3797108a028448946faR68

Then after inside handleCommand we should not send a command again if retries exceeded.